atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/metrics/TargetedSweepMetricPublicationFilter.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.metrics;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.LongSupplier;
+
+import org.immutables.value.Value;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.palantir.atlasdb.metrics.MetricPublicationFilter;
+
+/**
+ * Indicates whether targeted sweep metrics should be published.
+ *
+ * The criteria for publishing metrics for a given strategy, is if the number of enqueued writes or reads is in
+ * excess of {@link #MINIMUM_READS_WRITES_TO_BE_CONSIDERED_ACTIVE}, or it is believed that targeted sweep is behind by
+ * more than {@link #MINIMUM_STALE_DURATION}. If any of these conditions are true, then all targeted sweep metrics
+ * will be published.
+ */
+public class TargetedSweepMetricPublicationFilter implements MetricPublicationFilter {
+    @VisibleForTesting
+    static final long MINIMUM_READS_WRITES_TO_BE_CONSIDERED_ACTIVE = 1_000;
+    @VisibleForTesting
```

Comment:
Should we make this user-configurable?

Reply:
Dunno in this particular case, but in general, less config is always better. We should aim to build systems that either self-tune or have good enough defaults that work in most cases.

Reply:
Good question. I think I'll hang on to this for now: the kill switch allows us to recover our metrics if there's a P0 (or even a P1), and otherwise I doubt there'll be a situation where we need to change these so urgently that it can't wait for a quick PR that adds configurability.

Reply:
(In general I think I'd slant more towards leaving uncertain things configurable in AtlasDB, while less so in AtlasDB-Proxy mainly because of the relative difficulty/friction of recalling bad versions of AtlasDB.)